ordering/__init__.py

```diff
@@ -68,11 +68,13 @@ def __contains__(self, item: _T) -> bool:
         return item in self._labels
 
     def __iter__(self) -> Iterator[T]:
-        item = self._successors[self._start]
-
-        while not isinstance(item, _Sentinel):
+        successors = self._successors
+        item = successors[self._start]
+        end = self._end
+        while item is not end:
+            assert not isinstance(item, _Sentinel)
             yield item
-            item = self._successors[item]
+            item = successors[item]
```

Comment:
Not sure why you'd want to hide them from any child classes, but if that's how OrderedDict does it, I'm sure they've got a good reason.

I'm using `start` as a noun, not a verb, so `stop` doesn't really fit. If they were functions, I'd agree with you. I suppose an argument could be made for `first`, `last`, or `neg_inf`, `inf`, or something else, but I don't mind too much.